src/Engine/ProtoCore/Core.cs

```diff
@@ -770,7 +770,13 @@ public void GenerateExecutable()
         private int GetRuntimeTableSize()
         {
             // Due to the way this list is constructed, the largest id is the one of the last block.
-            return CompleteCodeBlockList.Last().codeBlockId + 1;
+            var lastBlock = CompleteCodeBlockList.Last();
+            // If the last block has children, then its last child has the largest id.
+            if (lastBlock.children.Count > 0)
+            {
+                lastBlock = lastBlock.children.Last();
+            }
+            return lastBlock.codeBlockId + 1;
```

Comment:
If the assertion that the last codeblock always has the largest id is true, then this function needs to be recursive.

Reply:
I have wondered about that too, but if you take a look at the part of the code I mention here https://github.com/DynamoDS/Dynamo/pull/10605#issuecomment-620108435, you'll see that only that the code doesn't do recursion, it also looks only at the first level of children. Maybe these code blocks can contain only one level in practice?

Reply:
The `BfsBuildSequenceTable` is recursive.

Reply:
@aparajit-pratap Yes, but it's using `CodeBlockList` which is actually a tree. Both this method and the one I pointed to in my previous comment use `CompleteCodeBlockList` which contains the same nodes but is flattened, to one level of children it would seem.

Reply:
There should be a codeblock for each language block (such as `[Imperative {...}]`, `if-else`, `while`, `for` etc., and function definition) so there can be any number of nesting or children of child codeblocks and so on.

Reply:
Ok, I thought so too that `CompleteCodeBlockList` was flattened. If that is confirmed, then I guess checking for one level is fine but I'm not quite sure why it is the last one that is always the largest.

Reply:
Could you provide an example where this is the case? It would definitely be nice to have a repro.

Reply:
> Ok, I thought so too that CompleteCodeBlockList was flattened. If that is confirmed, then I guess checking for one level is fine but I'm not quite sure why it is the last one that is always the largest.

I think I know why. If you take a look at the constructor of `CodeBlock`, the created code block is added immediately to the `CompleteCodeBlockList` using `Add`: `core.CompleteCodeBlockList.Add(this);`

Reply:
Then how can it be that the children can have the largest id? The largest should always be at the top level, i.e. `CompleteCodeBlockList`?

Reply:
I don't know if I understood what you meant. What I was saying is that the `List.Add` method adds to the end of the list, and since we always get a bigger id each time, by construction the biggest id should be at the end of the flattened list. Does that make sense?

Reply:
Yes, I understood that. If we're adding to the end of the `CompleteCodeBlockList`, we simply need to check the size of that list or the `codeBlockId` of the last item in `CompleteCodeBlockList`. Why are you checking for the last item in the `CompleteCodeBlockList`?

Reply:
> we simply need to check the size of that list or the codeBlockId of the last item in CompleteCodeBlockList

If we go back to the purpose of this PR, those two do not always match, so it's not the same to go with one or the other. Just wanted to make that clear.

> Why are you checking for children of the last item in the CompleteCodeBlockList?

By way of undo/redo in Josh's graph I was able to see a crash where the last code block contained children with larger ids than the id of its parent or the amount of elements in the list. It does seem strange that the children themselves were not part of the list, so it could be a byproduct of the undo/redo bug.

Reply:
I'm not so sure I understand the changes, to be honest. If you feel confident you are welcome to get this approved by someone else on the team. I'll do some investigation on my own to try to convince myself of your changes :)